aws_lambda_builders/workflows/java_gradle/gradlew_resolver.py

```diff
@@ -0,0 +1,29 @@
+"""
+Gradle executable resolution
+"""
+
+from aws_lambda_builders.path_resolver import PathResolver
+
+
+class GradlewResolver(object):
+    DUMMY_PATH = object()
+
+    def __init__(self, binary, executable_search_paths=None, path_resolver=None):
+        self.binary = binary
+        self.executables = [self.binary]
+        if path_resolver is None:
+            self.path_resolver = PathResolver(binary=self.binary, runtime=None,
+                                              executable_search_paths=executable_search_paths)
+        else:
+            self.path_resolver = path_resolver
+
+    @property
+    def exec_paths(self):
+        try:
+            return self.path_resolver.exec_paths
+        except ValueError as e:
+            # gradlew is optional so we're okay with not finding the executable
+            if str(e).startswith("Path resolution for runtime"):
```

Comment:
oh no! If you were to do hacks at this level, we shoudl either fix the Gradle implementation design or the Path resolver design.

Reply:
Providing `gradlew` is optional, but `PathResolver` throws a `ValueError` if it's not found.

Reply:
Thinking about it more, it will be cleaner to just create a custom path resolver for `gradlew` + `gradle`